linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java

```diff
@@ -206,14 +207,19 @@ static Model buildModelWithSnapshotBom(
     modelRequest.setProcessPlugins(false);
     modelRequest.setTwoPhaseBuilding(true); // This forces the builder stop after phase 1
     modelRequest.setPomFile(bomResult.getArtifact().getFile());
-    modelRequest.setModelResolver(new ProjectModelResolver(
-        session,
-        null,
-        repositorySystem,
-        new DefaultRemoteRepositoryManager(),
-        ImmutableList.of(),
-        null,
-        null));
+    modelRequest.setModelResolver(
+        new ProjectModelResolver(
+            session,
+            null,
+            repositorySystem,
+            new DefaultRemoteRepositoryManager(),
+            ImmutableList.of(CENTRAL), // Needed when parent pom is not locally available
+            null,
+            null));
+    // Profile activation needs JDK version through system properties
+    // https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/923
+    Properties modelBuildingProperties = modelRequest.getSystemProperties();
+    modelBuildingProperties.putAll(System.getProperties());
```

Comment:
testBuildModelWithSnapshotBom_JdkVersionActivation fails without this line due to the same error as #923